pyproject.toml

```
@@ -13,7 +13,7 @@ readme = "README.md"
 requires-python = ">=3.8"
 license = {text = "MIT"}
 dependencies = [
-  "ruff>=0.1.5, <0.2.0",
+  "ruff>=0.1.5, <0.3.0",
```

```suggestion
  "ruff>=0.2.0, <0.3.0",
```

Since 0.2.0 is the first version with a formatter enabled by default, as you said, this is required to go in hand with disabling the `yapf` and `autopep8` plugins.

---

Actually now that I think a bit - ruff formatting was available in 0.1.2, and in this plugin since before this MR. So I do believe we can keep the minimum version at 0.1.5 (as before)

ref: https://docs.astral.sh/ruff/formatter/

---

> Actually now that I think a bit - ruff formatting was available in 0.1.2

Yeah, but it was sort of in preview mode, wasn't it?

---

According to their docs, it is _still_ in `production-ready Beta`: https://docs.astral.sh/ruff/formatter/

I didn't do any particular changes in pylsp-ruff to enable formatting (prior to this PR)

---

I say we bump the ruff version to keep up-to-date with the latest ruff. I also see no harm in "forcing" people to use the newer ruff patch. We can roll back if this leads to issues for people.

---

Then I believe this is ready to be merged :-)